test/telemetry_metrics_statsd_test.exs

```diff
@@ -334,11 +334,9 @@ defmodule TelemetryMetricsStatsdTest do
 
       assert capture_log(fn ->
                TelemetryMetricsStatsd.udp_error(reporter, udp, :closed)
-               # errors.
-               eventually(fn ->
-                 {:ok, new_udp} = TelemetryMetricsStatsd.get_udp(pool_id)
-                 new_udp != udp
-               end)
+
+               # Wait for processing handle_cast by udp_error.
+               _ = TelemetryMetricsStatsd.get_pool_id(reporter)
```

Comment:
Ah, good catch. The reason this started flaking is because we introduced the default pool size of 10 _after_ this test was written.

It should all work if we set the `pool_size` of the reporter to 1 in this test. Then the UDP socket we get from `get_udp/1` will only be different when the reporter reopens a socket, which means it processed the error.

Reply:
@arkgil Thanks for your review!
Got it. I missed the `pool_size` option.

I revert the commit and set the `pool_size` to 1.
It passed. (See https://github.com/mopp/telemetry_metrics_statsd/pull/4)

Please check the latest commits 🙏